src/maintypes.jl

```diff
@@ -36,3 +36,31 @@ matrix internally, it is possible to use [`@manifold_element_forwards`](@ref) an
 [`@default_manifold_fallbacks`](@ref) to reduce implementation overhead.
 """
 abstract type AbstractManifoldPoint end
+
+"""
+    abstract type AbstractManifoldSize end
+
+Abstract representation of manifold size. Can be either [`StaticSize`](@ref) or
+[`RTSize`](@ref).
+"""
+abstract type AbstractManifoldSize end
+
+"""
+    StaticSize{T}
+
+Static size of a manifold.
+"""
+struct StaticSize{T} <: AbstractManifoldSize end
+StaticSize(t::NTuple) = StaticSize{t}()
+
+"""
+    RTSize{TS<:NTuple{N,Int} where N}
+
+Runtime size of a manifold. 
+"""
+struct RTSize{TS<:NTuple{N,Int} where {N}} <: AbstractManifoldSize
+    size::TS
+end
+
+getsize(::StaticSize{T}) where {T} = T
+getsize(S::RTSize) = S.size
```

Comment:
I am also not 100% sure about the `get_size_parameter(M)` method but checking that the field is zero in the implementation seems to me like a magic number and I would prefer a dispatch variant for accessing this, though in order to use it you have to implement it per manifold.
It will make code that uses the parameter more readable.

Reply:
Do you mean this: `return length(size) == 0 ? 1 : *(size...) * real_dimension(𝔽)` by "checking that the field is zero"? I can get rid of that using `prod` function.

Reply:
Yes that line has this “magic zero” I referred to.

Reply:
Latest commit removed it :slightly_smiling_face: